libraries/delivery_test_kitchen.rb

```diff
@@ -144,6 +146,65 @@ def prepare_kitchen_ec2
       file.run_action(:create)
     end
 
+  #
+  # Specific requirements for azurerm driver
+  #
+  # rubocop:disable AbcSize
+  # rubocop:disable Metrics/MethodLength
+  def prepare_kitchen_azurerm
+    fail 'Kitchen YAML file not found' unless kitchen_yaml?
+
+    # Load secrets from delivery-secrets data bag
+    secrets = get_project_secrets
+    fail 'Could not find secrets for kitchen-azurerm driver' \
+         ' in delivery-secrets data bag.' if secrets['azurerm'].nil?
+
+    # Variables used for configuring and running test kitchen EC2
+    cache                 = delivery_workspace_cache
+    azure_subscription_id = secrets['azurerm']['subscription_id']
+    azure_client_id       = secrets['azurerm']['client_id']
+    azure_client_secret   = secrets['azurerm']['client_secret']
+    azure_tenant_id       = secrets['azurerm']['tenant_id']
+    kitchen_instance_name = "test-kitchen-#{delivery_project}-#{delivery_change_id}"
+
+    @environment.merge!(
+      'AZURE_CLIENT_ID'            => azure_client_id,
+      'AZURE_CLIENT_SECRET'        => azure_client_secret,
+      'AZURE_TENANT_ID'            => azure_tenant_id,
+      'KITCHEN_INSTANCE_NAME'      => kitchen_instance_name
+    )
+
+    # Installing kitchen-azurerm driver
+    begin
+      chef_gem = Chef::Resource::ChefGem.new('kitchen-azurerm', run_context)
+      chef_gem.run_action(:install)
+    rescue => e
+      Chef::Log.error('Azure gem installation failed. You might need to ensure that dev tools are installed.')
+      Chef::Log.error("Add depends 'build-essential', '~> 7.0.2' to the metadata.rb of your build cookbook and add include_recipe 'build-essential::default' to the default.rb of your build cookbook")
```

Comment:
Fantastic! Thank you for catching the error and the nice message, two things. Maybe we dont need to be specific with the build-essentials version. Also the line is to long 😝  this will fail in lint when we push it through Automate...